hooks/post_gen_project.py

```diff
@@ -497,8 +505,52 @@ def main():
     if "{{ cookiecutter.use_async }}".lower() == "n":
         remove_async_files()
 
+    setup_dependencies()
+
     print(SUCCESS + "Project initialized, keep up the good work!" + TERMINATOR)
 
 
+def setup_dependencies():
+    print("Installing python dependencies using uv...")
+
+    if "{{ cookiecutter.use_docker }}".lower() == "y":
+        # Build the Docker service using Docker Compose
+        try:
+            subprocess.run(["docker", "compose", "-f", "docker-compose.local.yml", "build", "django"], check=True)
+        except subprocess.CalledProcessError as e:
+            print(f"Error building Docker service: {e}", file=sys.stderr)
+            sys.exit(1)
+
+        # Use Docker to run the uv command
+        uv_cmd = ["docker", "compose", "-f", "docker-compose.local.yml", "run", "--rm", "django", "uv"]
```

Comment:
Apologies if this was already discussed, I didn't follow the latest changes too closely. Why do we need to run uv in Docker? Why not run the uv commands on the host?

No need to change anything, I'm curious to understand what made you opt for this version...

Reply:
You need to run uv inside your environment, otherwise it will install Python and all the other packages outside of the container.

Reply:
Something I'm wondering (as a potential improvement after we merge this), would it be worth providing a more lightweight image just to run this? Feels like an "expensive" step just to add the deps... Again, no need to change anything here, and also I have not tried it yet

Reply:
Yes, you are right, we only need to calculate the dependency tree here and write it down. We do not need compiling psycopg for example at this stage. But 'uv add' does both of them and if libpg-dev isn't available pyproject.toml will not be updated. 

I didn't researched it any further, so maybe uv has a flag for not installing things but I don't know.

Reply:
Got it. Yes no problem, we can optimise later

Reply:
Noticed a severe slowdown in the Github actions due to this. Adding `--no-sync` to `uv add` seems to be the solution.